### PR TITLE
[Component] Progress Elements

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Fixes https://github.com/brave/leo/issues/67
Fixes https://github.com/brave/leo/issues/66

Adds a progress ring and a progress bar. In addition, this PR makes it possible to have multiple components in a single folder.

### CODE REVIEW

**Review by aguscruiz — APPROVED** [review]
Awesome! I love how you can setup different widths and strokes. Amazing!

**Review by AlanBreck — APPROVED** [review]
It's beautiful.

File: rollup.config.js [hunk 1]
```diff
@@ -8,14 +8,15 @@ import fs from 'fs'
...
 // components.
 const COMPONENTS_FOLDER = path.resolve('./', 'src', 'components')
-const inputs = fs
-  .readdirSync(COMPONENTS_FOLDER, { withFileTypes: true })
-  .filter((f) => f.isDirectory())
-  .map((f) => path.join(COMPONENTS_FOLDER, `${f.name}/${f.name}.svelte`))
+const inputs = []
+
+for await (const file of getSvelteFiles(COMPONENTS_FOLDER, false))
+  inputs.push(file)
```
Comment (AlanBreck) [inline rollup.config.js:19]: My only nit would be that `{}` brackets around the `for` loop block would be nice. ;-)

Comment (fallaciousreasoning) [reply rollup.config.js]: Booooo :laughing: (I've added them for you)

Comment (AlanBreck) [reply rollup.config.js]: Ha ha! Thank you. :-)

File: src/components/progress/progressRing.svelte [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  $: circumference = normalizedRadius * 2 * Math.PI
+</script>
+
+<div
+  class="leo-progressring"
+  class:spin={mode === 'indeterminate'}
+  style="width: {radius * 2}px; height: {radius * 2}px"
+>
+  <slot />
+  <svg width={radius * 2} height={radius * 2}>
```
Comment (petemill) [inline src/components/progress/progressRing.svelte:21]: Wondering whether we can set this to a 24x24 viewbox, and always set the radius of the circle to 24, then allow width / height of this component to be changed via css property, similar (or same as) `--leo-icon-size` @fallaciousreasoning

**Pull request merged**